tests/main/snap-quota-thread/task.yaml

```
@@ -27,6 +27,21 @@ execute: |
   # count of threads expected is 33 including the main thread.
   SPAWNER_MAX_THREADS=33
 
+  function wait_for_thread_usage() {
+    local threadCount=0
+    local runs=$1
+    local expected=$2
+
+    for _ in $(seq "${runs}"); do
+        threadCount="$(snap quota group-one | awk '/current:/,/threads:/{printf "%s",$2} END {print ""}')"
```

I suppose this should work too without resorting to END trickery with newlines
```suggestion
        threadCount="$(snap quota group-one | awk '/current:/,/threads:/{if ($1 ~ "threads:") { print $2 } }'
```

---

nice! added